Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_common.py

```diff
@@ -102,15 +170,250 @@ def draw(self, context):
+        return {'FINISHED'}
+
+class MESHSYNC_OT_CheckProjectPath_Manual(bpy.types.Operator):
+    bl_idname = "meshsync.check_project_path_manual"
+    bl_label = "Ensure Unity project path is pointing to a Unity project"
+    
+    def execute(self, context):
+        global click_source
+        click_source = 'Manual'
+        bpy.ops.meshsync.check_project_path('INVOKE_DEFAULT')
+        return {'FINISHED'}
+
+
+class MESHSYNC_OT_CheckProjectPath(bpy.types.Operator):
+    bl_idname = "meshsync.check_project_path"
+    bl_label = "Ensure Unity project path is pointing to a Unity project"
+
+    def execute(self, context):
+        directory = context.scene.meshsync_unity_project_path
+        #Validate that the path is a project
+        if msb_validate_project_path(directory) == False:
+            #Try to fill a valid path from an already running project
+            bpy.ops.meshsync.path_from_server('INVOKE_DEFAULT')
+        else:
+            bpy.ops.meshsync.install_meshsync('INVOKE_DEFAULT')
+
+        return {'FINISHED'}
+
+class MESHSYNC_OT_TryGetPathFromServer(bpy.types.Operator):
+    bl_idname = "meshsync.path_from_server"
+    bl_label = "Try to get the unity path from a running server"
+
+    def execute(self, context):
+        if not msb_context.is_editor_server_available:
+            bpy.ops.meshsync.prompt_project_path('INVOKE_DEFAULT')
+            return {'FINISHED'}
+
+        #Get the project path
+        msb_context.sendEditorCommand(2)
+        context.scene.meshsync_unity_project_path = msb_context.editor_command_reply
+        directory = context.scene.meshsync_unity_project_path
+
+        if msb_validate_project_path(directory) == False:
+            # Prompt user to enter a valid path
+            bpy.ops.meshsync.prompt_project_path('INVOKE_DEFAULT')
+        else:
+            bpy.ops.meshsync.create_server('INVOKE_DEFAULT')
+            
+        return {'FINISHED'}
+
+class MESHSYNC_OT_PromptProjectPath(bpy.types.Operator, ImportHelper):
+    bl_idname = "meshsync.prompt_project_path"
+    bl_label = "Unity Project"
+
+    filepath : bpy.props.StringProperty(name = "filepath")
+
+    def execute(self, context):
+
+        context.scene.meshsync_unity_project_path = self.filepath
+
+        if msb_validate_project_path(self.filepath) == False:
+            message = self.filepath + " is not a Unity project"
+            MS_MessageBox(message = message)
+            print(message)
+            return {'FINISHED'}
+        else:
+            bpy.ops.meshsync.install_meshsync('INVOKE_DEFAULT')
+
+        return {'FINISHED'}
+
+class MESHSYNC_OT_InstallMeshSync(bpy.types.Operator):
+
+    bl_idname = "meshsync.install_meshsync"
+    bl_label = "Try to get the unity path from a running server"
+
+    def get_meshsync_path(self):
+        script_file = os.path.realpath(__file__)
+        directory = os.path.dirname(script_file)
+        system = platform.system()
+        if system == 'Windows':
```

Comment:
Aren't we already modifying the manifest so that the project downloads a particular MeshSync version from the Internet ?

If so, why are we bundling a particular version on MeshSync in the plugin ?

This will make the size of MeshSyncDCCPlugins a lot bigger.
Probably around (15 MB * number of supported Blender versions) ?

Reply:
Good call, I will change this

Reply:
done, currently set to 0.14.5-preview but will need to be updated with the next version that will contain the server changes